src/nbl/ext/ImGui/ImGui.cpp

```diff
@@ -283,19 +504,19 @@ namespace nbl::ext::imgui
 				auto& param = params[0u];
 				param.layout = pipelineLayout.get();
 				param.shaders = specs;
-				param.renderpass = m_creationParams.renderpass;
-				param.cached = { .vertexInput = vertexInputParams, .primitiveAssembly = primitiveAssemblyParams, .rasterization = rasterizationParams, .blend = blendParams, .subpassIx = m_creationParams.subpassIx };
+				param.renderpass = creationParams.renderpass.get();
+				param.cached = { .vertexInput = vertexInputParams, .primitiveAssembly = primitiveAssemblyParams, .rasterization = rasterizationParams, .blend = blendParams, .subpassIx = creationParams.subpassIx };
 			};
 			
-			if (!m_creationParams.utilities->getLogicalDevice()->createGraphicsPipelines(m_creationParams.pipelineCache, params, &pipeline))
+			if (!creationParams.utilities->getLogicalDevice()->createGraphicsPipelines(creationParams.pipelineCache.get(), params, &m_pipeline))
 			{
-				m_creationParams.utilities->getLogger()->log("Could not create pipeline!", system::ILogger::ELL_ERROR);
+				creationParams.utilities->getLogger()->log("Could not create pipeline!", ILogger::ELL_ERROR);
 				assert(false);
 			}
 		}
```

Comment:
would prefer to return the pipeline or something then nullptr return as the error behaviour, think about how nasty it  must be to have the extension assert-crash but only in debug mode.

Reply:
resolved